funcx_endpoint/funcx_endpoint/endpoint/endpoint_manager.py

```diff
@@ -0,0 +1,365 @@
+        else:
+            stdout = sys.stdout
+            stderr = sys.stderr
+
+        try:
+            context = daemon.DaemonContext(working_directory=endpoint_dir,
+                                           umask=0o002,
+                                           pidfile=daemon.pidfile.PIDLockFile(
+                                               os.path.join(endpoint_dir, 'daemon.pid')),
+                                           stdout=stdout,
+                                           stderr=stderr,
+                                           detach_process=self.endpoint_config.config.detach_endpoint)
+
+        except Exception:
+            self.logger.exception("Caught exception while trying to setup endpoint context dirs")
+
+        self.check_pidfile(context.pidfile.path, "funcx-endpoint")
+
+        with context:
+            self.daemon_launch(funcx_client, endpoint_uuid, endpoint_dir, keys_dir)
+
+    def daemon_launch(self, funcx_client, endpoint_uuid, endpoint_dir, keys_dir):
+        # Register the endpoint
+        self.logger.info("Registering endpoint")
+        reg_info = retry_call(self.register_endpoint, fargs=[funcx_client, endpoint_uuid, endpoint_dir], delay=10, max_delay=300, backoff=1.2)
+        self.logger.info("Endpoint registered with UUID: {}".format(reg_info['endpoint_id']))
+
+        # Configure the parameters for the interchange
+        optionals = {}
+        optionals['client_address'] = reg_info['public_ip']
+        optionals['client_ports'] = reg_info['tasks_port'], reg_info['results_port'], reg_info['commands_port'],
+        if 'endpoint_address' in self.funcx_config:
+            optionals['interchange_address'] = self.funcx_config['endpoint_address']
+
+            optionals['logdir'] = endpoint_dir
+
+        if self.DEBUG:
+            optionals['logging_level'] = logging.DEBUG
+
+        ic = EndpointInterchange(self.endpoint_config.config,
+                                 endpoint_id=endpoint_uuid,
+                                 keys_dir=keys_dir,
+                                 **optionals)
+        ic.start()
+        ic.stop()
+
+        self.logger.critical("Interchange terminated.")
+
+    # Avoid a race condition when starting the endpoint alongside the web service
+    def register_endpoint(self, funcx_client, endpoint_uuid, endpoint_dir):
+        """Register the endpoint and return the registration info.
+
+        Parameters
+        ----------
+
+        funcx_client : FuncXClient
+            The auth'd client to communicate with the funcX service
+
+        endpoint_uuid : str
+            The uuid to register the endpoint with
+
+        endpoint_dir : str
+            The directory to write endpoint registration info into.
+
+        """
+        self.logger.debug("Attempting registration")
+        self.logger.debug(f"Trying with eid : {endpoint_uuid}")
+        import funcx_endpoint
```

Comment:
Why is this import here? 
PEP 08 states:
Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.

Reply:
Moved it to the top